core/src/scene/sceneLoader.cpp

```diff
@@ -722,12 +722,25 @@ void SceneLoader::loadCameras(Node _cameras, Scene& _scene) {
 
             view->setCameraType(CameraType::perspective);
 
-            if (Node fov = camera["fov"]) {
-                // TODO
-            }
-
+            // Only one of focal length and FOV is applied;
+            // according to docs, focal length takes precedence.
             if (Node focal = camera["focal_length"]) {
-                // TODO
+                if (focal.IsScalar()) {
+                    float length = focal.as<float>(view->getFocalLength());
```

Comment:
It attempts to decode the node to the prescribed type and returns the fallback if either the node is invalid or the decoding fails: https://github.com/jbeder/yaml-cpp/blob/master/include/yaml-cpp/node/impl.h#L85-L99

I prefer this because it's more concise, doesn't involve exceptions, and works for any value that supports decoding.